packages/opentelemetry-exporter-prometheus/src/PrometheusExporter.ts

```diff
@@ -168,7 +178,7 @@ export class PrometheusExporter implements MetricExporter {
     request: IncomingMessage,
     response: ServerResponse
   ) => {
-    if (url.parse(request.url!).pathname === this._endpoint) {
+    if (request.url && new url.URL(request.url).pathname === this._endpoint) {
```

Comment:
`url.URL` throws in quite some cases where `url.parse` didn't.
Besides that `request.url` is usually a relative url (path + query) therefore extra care is needed here. see https://nodejs.org/dist/latest-v15.x/docs/api/http.html#http_message_url and 
https://github.com/nodejs/node/issues/35458

Reply:
Thanks!

As per the comments in the issue you linked, seems like `url.parse()` is much more straightforward despite the deprecation warning. Reverted!